### PR TITLE
Add isNamespaced flag to values.yaml and update related templates and code

### PR DESCRIPTION
**Pull request by Bezbran**
## **Type**
Enhancement, Tests

___

## **Description**
- Added support for running the application in a namespaced environment. This is controlled by a new flag `isNamespaced` in `values.yaml`.
- Updated the main application and the rule binding store to use a new environment variable `STORE_NAMESPACE` to store the namespace.
- Updated the Helm templates to support running in a namespaced environment.
- Added a new GitHub Actions workflow for running system tests in a namespaced environment.
- Updated the tests for the rule binding store to reflect the changes.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>main.go</strong><dd><code>Added namespace support in main.go</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

cmd/main.go
- Added a new environment variable `STORE_NAMESPACE` to store the <br>  namespace.
- Updated the `NewRuleBindingK8sStore` function call to include <br>  `storeNamespace`.
    </details>
  </td>
  <td><a href="https://github.com/armosec/kubecop/pull/164/files#diff-c444f711e9191b53952edb65bfd8c644419fc7695c62611dc0fb304b4fb197d6">+7/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>store.go</strong><dd><code>Added namespace support in rule binding store</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pkg/rulebindingstore/store.go
- Added `storeNamespace` to the `RuleBindingK8sStore` struct.
- Updated the `NewRuleBindingK8sStore` function to accept <br>  `storeNamespace` as an argument.
- Updated the `StartController` function to use `storeNamespace` when <br>  creating the informer.
    </details>
  </td>
  <td><a href="https://github.com/armosec/kubecop/pull/164/files#diff-4c19cf4bcd9fac5ffcf44014b76e894e889d5a0dd2319e72218064ff96feae55">+7/-2</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    
</table></details></td></tr><tr><td><strong>Tests</strong></td><td><details><summary>3 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>store_test.go</strong><dd><code>Updated tests for rule binding store</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pkg/rulebindingstore/store_test.go
- Updated the `NewRuleBindingK8sStore` function calls in the tests to <br>  include an empty string for `storeNamespace`.
    </details>
  </td>
  <td><a href="https://github.com/armosec/kubecop/pull/164/files#diff-2dd20e1423f09d7d4dd81983c2fe41f40a72904fa576b97a0dd7e4a8c711e85d">+4/-4</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ci.yaml</strong><dd><code>Added validation for namespaced Helm template in CI workflow</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

.github/workflows/ci.yaml
- Added a new step to validate the namespaced Helm template.
    </details>
  </td>
  <td><a href="https://github.com/armosec/kubecop/pull/164/files#diff-944291df2c9c06359d37cc8833d182d705c9e8c3108e7cfe132d61a06e9133dd">+4/-0</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>namespaced-system-tests.yaml</strong><dd><code>Added namespaced system tests workflow</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

.github/workflows/namespaced-system-tests.yaml
- Added a new GitHub Actions workflow for running system tests in a <br>  namespaced environment.
    </details>
  </td>
  <td><a href="https://github.com/armosec/kubecop/pull/164/files#diff-416d06049e4f75d9aa88496af3a430ffb9bfc83bcc1489914a70cff1035b5789">+149/-0</a>&nbsp; </td>
</tr>                    
</table></details></td></tr><tr><td><strong>Configuration changes</strong></td><td><details><summary>5 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>runtime-rule-binding.crd.yaml</strong><dd><code>Added namespaced support in CRD</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

chart/kubecop/crds/runtime-rule-binding.crd.yaml
- Added conditional logic to set the scope of the CRD to `Namespaced` if <br>  `.Values.isNamespaced` is true.
    </details>
  </td>
  <td><a href="https://github.com/armosec/kubecop/pull/164/files#diff-fd74bc23a77dded82ef08a1d7d69295133cbed6605700fdf3a46708336b0dc31">+4/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>controller-statefulset.yaml</strong><dd><code>Added namespaced support in controller statefulset template</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

chart/kubecop/templates/controller-statefulset.yaml
- Added a new environment variable `STORE_NAMESPACE` that gets its value <br>  from the pod's namespace if `.Values.isNamespaced` is true.
    </details>
  </td>
  <td><a href="https://github.com/armosec/kubecop/pull/164/files#diff-94879afdd23579720c115da5b7271da861315680ff43c4546e9ef87b19e35c9b">+6/-0</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>deamonset.yaml</strong><dd><code>Added namespaced support in daemonset template</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

chart/kubecop/templates/deamonset.yaml
- Added a new environment variable `STORE_NAMESPACE` that gets its value <br>  from the pod's namespace if `.Values.isNamespaced` is true.
    </details>
  </td>
  <td><a href="https://github.com/armosec/kubecop/pull/164/files#diff-85e96503359a8117f75f7b5e72b88e65cad1abdd451b7ea40e1c199289434d39">+6/-0</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>default-rule-binding.yaml</strong><dd><code>Added namespaced support in default rule binding template</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

chart/kubecop/templates/default-rule-binding.yaml
- Added the namespace of the release to the metadata of the default rule <br>  binding if `.Values.isNamespaced` is true.
    </details>
  </td>
  <td><a href="https://github.com/armosec/kubecop/pull/164/files#diff-d07b435100ee64117d45d4d5fa6d3554abc7e3647f1663729ca54a53bd33f088">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>values.yaml</strong><dd><code>Added namespaced flag in values.yaml</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

chart/kubecop/values.yaml
- Added a new value `isNamespaced` and set it to true.
    </details>
  </td>
  <td><a href="https://github.com/armosec/kubecop/pull/164/files#diff-a76a9e679fe402f7217b3d2cef96f308de47dc05a8d4cd9b8920a9bf051e0180">+2/-0</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    
</table></details></td></tr></tr></tbody></table><hr>

<details> <summary><strong>✨ Usage guide:</strong></summary><hr> 

**Overview:**
The `describe` tool scans the PR code changes, and generates a description for the PR - title, type, summary, walkthrough and labels. The tool can be triggered [automatically](https://github.com/Codium-ai/pr-agent/blob/main/Usage.md#github-app-automatic-tools) every time a new PR is opened, or can be invoked manually by commenting on a PR.

When commenting, to edit [configurations](https://github.com/Codium-ai/pr-agent/blob/main/pr_agent/settings/configuration.toml#L46) related to the describe tool (`pr_description` section), use the following template:
```
/describe --pr_description.some_config1=... --pr_description.some_config2=...
```
With a [configuration file](https://github.com/Codium-ai/pr-agent/blob/main/Usage.md#working-with-github-app), use the following template:
```
[pr_description]
some_config1=...
some_config2=...
```

<table><tr><td><details> <summary><strong> Enabling\disabling automation </strong></summary><hr>

- When you first install the app, the [default mode](https://github.com/Codium-ai/pr-agent/blob/main/Usage.md#github-app-automatic-tools) for the describe tool is:
```
pr_commands = ["/describe --pr_description.add_original_user_description=true" 
                         "--pr_description.keep_original_user_title=true", ...]
```
meaning the `describe` tool will run automatically on every PR, will keep the original title, and will add the original user description above the generated description. 

- Markers are an alternative way to control the generated description, to give maximal control to the user. If you set:
```
pr_commands = ["/describe --pr_description.use_description_markers=true", ...]
```
the tool will replace every marker of the form `pr_agent:marker_name` in the PR description with the relevant content, where `marker_name` is one of the following:
  - `type`: the PR type.
  - `summary`: the PR summary.
  - `walkthrough`: the PR walkthrough.

Note that when markers are enabled, if the original PR description does not contain any markers, the tool will not alter the description at all.
        

</details></td></tr>

<tr><td><details> <summary><strong> Custom labels </strong></summary><hr>

The default labels of the `describe` tool are quite generic: [`Bug fix`, `Tests`, `Enhancement`, `Documentation`, `Other`].

If you specify [custom labels](https://github.com/Codium-ai/pr-agent/blob/main/docs/DESCRIBE.md#handle-custom-labels-from-the-repos-labels-page-gem) in the repo's labels page or via configuration file, you can get tailored labels for your use cases.
Examples for custom labels:
- `Main topic:performance` - pr_agent:The main topic of this PR is performance
- `New endpoint` - pr_agent:A new endpoint was added in this PR
- `SQL query` - pr_agent:A new SQL query was added in this PR
- `Dockerfile changes` - pr_agent:The PR contains changes in the Dockerfile
- ...

The list above is eclectic, and aims to give an idea of different possibilities. Define custom labels that are relevant for your repo and use cases.
Note that Labels are not mutually exclusive, so you can add multiple label categories.
Make sure to provide proper title, and a detailed and well-phrased description for each label, so the tool will know when to suggest it.        

</details></td></tr>

<tr><td><details> <summary><strong> Inline File Walkthrough 💎</strong></summary><hr>

For enhanced user experience, the `describe` tool can add file summaries directly to the "Files changed" tab in the PR page.
This will enable you to quickly understand the changes in each file, while reviewing the code changes (diffs).

To enable inline file summary, set `pr_description.inline_file_summary` in the configuration file, possible values are:
- `'table'`: File changes walkthrough table will be displayed on the top of the "Files changed" tab, in addition to the "Conversation" tab.
- `true`: A collapsable file comment with changes title and a changes summary for each file in the PR.
- `false` (default): File changes walkthrough will be added only to the "Conversation" tab.
<tr><td><details> <summary><strong> Utilizing extra instructions</strong></summary><hr>

The `describe` tool can be configured with extra instructions, to guide the model to a feedback tailored to the needs of your project.

Be specific, clear, and concise in the instructions. With extra instructions, you are the prompter. Notice that the general structure of the description is fixed, and cannot be changed. Extra instructions can change the content or style of each sub-section of the PR description.

Examples for extra instructions:
```
[pr_description] 
extra_instructions="""
- The PR title should be in the format: '<PR type>: <title>'
- The title should be short and concise (up to 10 words)
- ...
"""
```
Use triple quotes to write multi-line instructions. Use bullet points to make the instructions more readable.

</details></td></tr>

<tr><td><details> <summary><strong> More PR-Agent commands</strong></summary><hr> 

> To invoke the PR-Agent, add a comment using one of the following commands:  
> - **/review**: Request a review of your Pull Request.   
> - **/describe**: Update the PR title and description based on the contents of the PR.   
> - **/improve [--extended]**: Suggest code improvements. Extended mode provides a higher quality feedback.   
> - **/ask \<QUESTION\>**: Ask a question about the PR.   
> - **/update_changelog**: Update the changelog based on the PR's contents.   
> - **/add_docs** 💎: Generate docstring for new components introduced in the PR.   
> - **/generate_labels** 💎: Generate labels for the PR based on the PR's contents.   
> - **/analyze** 💎: Automatically analyzes the PR, and presents changes walkthrough for each component.   

>See the [tools guide](https://github.com/Codium-ai/pr-agent/blob/main/docs/TOOLS_GUIDE.md) for more details.
>To list the possible configuration parameters, add a **/config** comment.   
 

</details></td></tr>

</table>

See the [describe usage](https://github.com/Codium-ai/pr-agent/blob/main/docs/DESCRIBE.md) page for a comprehensive guide on using this tool.

</details>

### CODE REVIEW

**Comment by amitschendel** [comment]
@Bezbran Can you modify it to not be "store" specific but also applies to application profiles? Like, not just for rule bindings because I want to use this namespace also for application profiles storing.

**Comment by Bezbran** [comment]
> @Bezbran Can you modify it to not be "store" specific but also applies to application profiles? Like, not just for rule bindings because I want to use this namespace also for application profiles storing.

I use "STORE_" to emphasize this is for all objects, not only binding...

File: .github/workflows/namespaced-system-tests.yaml [hunk 1]
```diff
@@ -0,0 +1,149 @@
+name: KubeCop namespaced System Test
+
+on:
+  pull_request:
+    paths-ignore:
+      - '**/*.md'
+    types: [labeled, synchronize, ready_for_review, opened, reopened]
+
```
[the remark this answers is not in the document]
Comment (Bezbran) [reply .github/workflows/namespaced-system-tests.yaml:8]: I introduced a new label for it

[the remark this answers is not in the document]
Comment (Bezbran) [reply .github/workflows/namespaced-system-tests.yaml:8]: [requires-namespaced-system-test](https://github.com/armosec/kubecop/labels/requires-namespaced-system-test)

[the remark this answers is not in the document]
Comment (Bezbran) [reply .github/workflows/namespaced-system-tests.yaml:8]: Since we are installing the environment in the GHA and not in the tests framework

**Review by slashben — APPROVED** [review]

**Pull request merged**